clustermesh-apiserver/main.go

```diff
@@ -197,7 +197,7 @@ func runApiserver() error {
 	flags.String(option.IdentityAllocationMode, option.IdentityAllocationModeCRD, "Method to use for identity allocation")
 	option.BindEnv(option.IdentityAllocationMode)
 
-	flags.IntVar(&clusterID, option.ClusterIDName, 0, "Cluster ID")
```

Comment:
cluster id value must be from 1-255 only, what do you think about having the type as uint8 ?

uint32 is a equivalent of int type, at least for intention, above is just a question.

Reply:
Yeah, I first thought that, but my thought here is that 1-255 is just coming from the limitation of the DPlane side (bit width of the numeric identity) and nothing prevents us from using the larger value in CPlane. I think that's why we are using `int` for CPlane code right now.

Thinking about the future extension, I'm afraid of limiting the ClusterID based on the type like by using `uint8`, because once we expose it as a user-facing interface and the interface is aware of the bit width (like protobuf, or open API with explicit bit width specification, depends on their wire format), it is hard to change it. Currently, there's no such code, but in the future, I'm planning to expose ClusterID as a parameter of one of the cilium-agent's APIs.

By these reason, I prefer to use `uint32` instead of `uint8`, what do you think?

Reply:
oh right, you are already thinking about it (as expected), thanks :100: